### PR TITLE
Make uint256 and uint160 IComparable

### PR DESCRIPTION
**Pull request by joemphilips**
I need this for using uint256 as a key for F#'s `Map` type.

### CODE REVIEW

**Comment by nopara73** [comment]
This can be also useful to eventually be able to compare and order `OutPoint`s with each other based on @kristovatlas's [BIP69.](https://github.com/bitcoin/bips/blob/master/bip-0069.mediawiki) I'm not sure we have to go that far anytime soon, maybe if someone needs it, but it's a good stepping stone.

File: NBitcoin/UInt2561.cs [hunk 1]
```diff
@@ -477,8 +477,25 @@ public override int GetHashCode()
 			}
 			return hash;
 		}
+
+		public int CompareTo(object other)
+		{
+			var o = other as uint256;
+			if (o == null)
+				return -1;
+			return CompareTo(o);
```
Comment (nopara73) [inline NBitcoin/UInt2561.cs]: No need to do the casting before null check.

```cs
if(other is null)
	return -1;
var o = other as uint256;
return CompareTo(o);
```

File: NBitcoin/UInt2561.cs [hunk 2]
```diff
@@ -819,5 +836,20 @@ public override int GetHashCode()
 			}
 			return hash;
 		}
+		public int CompareTo(object other)
+		{
+			var o = other as uint160;
+			if (o == null)
+				return -1;
+			return CompareTo(o);
```
Comment (nopara73) [inline NBitcoin/UInt2561.cs]: Same here.

File: NBitcoin/UInt2561.cs [hunk 3]
```diff
@@ -477,8 +477,25 @@ public override int GetHashCode()
...
+
+		public int CompareTo(object other)
+		{
+			if (other is null)
+				return -1;
+			var o = other as uint256;
+			return CompareTo(o);
+		}
+
+		public int CompareTo(uint256 other)
```
Comment (NicolasDorier) [inline NBitcoin/UInt2561.cs]: You can treat the `other is null` case here

**Comment by joemphilips** [comment]
closing since #697 has done the same.

**Pull request closed without merging**